unit_tests/aw_test_utils/iter_tools.H

```diff
@@ -48,6 +48,34 @@ void run_algorithm(amr_wind::Field& field, const Functor& func)
     }
 }
 
+template <typename Functor>
```

Comment:
since you added these, can we make the other ones (above) call these? That would remove a bunch of duplicate code.

Reply:
Not sure what you mean here. I added these so that run_algorithm could operate on integer fields. How would the above ones (real fields) use these?

Reply:
Sorry totally not clear. I couldn't even parse what I said a week ago. Here's another stab ;) We have 2 versions or run_algorithm: 1 with a for loop on levels, another for 1 level (ignoring that we need one for int fields and another for real fields). I would think we could:

```
void run_algorithm(
    const int nlevels,
    amrex::Vector<amrex::iMultiFab*>& intfield,
    const Functor& func)
{
    for (int lev = 0; lev < nlevels; ++lev) {
        run_algorithm(intfield, funct);
    }
}
```

And do that for the int and real versions.

Reply:
even better ;)